vyper/venom/README.md

```diff
@@ -160,3 +160,257 @@ A number of passes that are planned to be implemented, or are implemented for im
+An operand can be a label, a variable, or a literal.
+
+## Instructions
+
+### Special instructions
+
+- `invoke`
+  - Cause control flow to jump to a function denoted by the label.
+  - Return values are passed in the return buffer at the offset address.
+  - Practically only used for internal functions.
+  - Effectively translates to `JUMP` and therefore changes the program counter value.
+  - ```
+    invoke offset, label
+    ```
+- `alloca`
+  - Allocates memory of a given size at a given offset in memory.
+  - The output is the offset itself.
+  - Because the SSA form does not allow changing values of registers, handling mutable variables can be tricky. The `alloca` instruction is meant to simplify that.
+  - ```
+    out = alloca size, offset
+    ```
+- `palloca`
+  - Like the `alloca` instruction but only used for parameters of internal functions.
+  - ```
+    out = palloca size, offset
+    ```
+- `iload`
+  - Load value at immutable section of memory denoted by `offset` into `out` variable.
+  - The operand can be either a literal, which is a statically computed offset, or a variable.
+  - ```
+    out = iload offset
+    ```
+- `istore`
+  - The instruction represents a store into immutable section of memory.
+  - Like in `iload`, the offset operand can be a literal.
+  - ```
+    istore offset value
+    ```
+- `phi`
+  - Because in SSA form each variable is assigned just once, it is tricky to handle that variables may be assigned to something different based on which program path was taken.
+  - Therefore, we use `phi` instructions. They are used in basic blocks where the control flow path merges.
+  - So essentially the `out` variable is set to `var_a` if the program entered this block from `label_a` or to `var_b` when it went through `label_b`. 
+  - ```
+    out = phi var_a, label_a, var_b, label_b
+    ```
+- `offset`
+  - Statically compute offset. Useful for `mstore`, `mload` and such.
+  - Basically `label` + `op`.
+  - ```
+    ret = offset label, op
+    ```
+- `param`
+  - The `param` instruction is used to represent function arguments passed by the stack.
+  - We assume the argument is on the stack and the `param` instruction is used to ensure we represent the argument by the `out` variable.
+  - ```
+    out = param
+    ```
+- `store`
+  - Store variable value or literal into `out` variable.
+  - ```
+    out = op
+    ```
+- dbname
+  - make and mark a data segment (one data segment in context - so maybe section it?) dunno
+- db
+  - db stores into the data segment some label? hmm
+- `dloadbytes`
+  - Alias for `codecopy` for legacy reasons. May be removed in future versions.
+- `ret`
+  - Represents a return from an internal call.
+  - Jumps to a location given by `op`, hence modifies the program counter.
+  - ```
+    ret op
+    ```
+- `exit`
+  - Similar to `stop`, but used for constructor exit. The assembler is expected to jump to a special initcode sequence which returns the runtime code.
```

Comment:
Do you know why it may be in fallback? there is the revert before it anyway, but it confused me as it doesn't seem to do much constructor exit. 
```fallback:  IN=[__main_entry, 47_if_exit] OUT=[] => {}
    revert 0, 0
    exit```

Reply:
I don't think it should be in runtime code...

Reply:
I think it is:  ```--experimental-codegen  -f bb_runtime``` on 
```
totalShares: public(uint256)

# Set up the company.
@deploy
def __init__(_total_shares: uint256):
    pass
```

Reply:
yeah i see what you mean. dead code! but it shouldn't be there, i would consider that a bug in our venom generation.